pkg/db/library.go

```diff
@@ -39,58 +46,81 @@ func StorePaths(givenLibraries []config.Library) error {
 }
 
 func GetOnlyLibraries() ([]model.Library, error) {
-	stmt := SELECT(Library.AllColumns).FROM(Library.Table)
 	var libraries []model.Library
+	err := database.QB().
+		From("library").
+		ScanStructs(&libraries)
 
-	err := stmt.Query(db(), &libraries)
 	return libraries, err
 }
 
 func GetLibraries() ([]CombinedLibrary, error) {
-	stmt := SELECT(Library.AllColumns, Gallery.AllColumns).
-		FROM(Library.LEFT_JOIN(Gallery, Gallery.LibraryID.EQ(Library.ID)))
 	var libraries []CombinedLibrary
+	err := database.QB().
+		From("library").
+		LeftJoin(
+			goqu.T("gallery"),
+			goqu.On(goqu.Ex{
+				"gallery.id": goqu.I("library.id"),
+			}),
+		).
+		ScanStructs(&libraries)
 
-	err := stmt.Query(db(), &libraries)
 	return libraries, err
 }
 
 // getLibrary returns the library from the database based on the ID or path.
 func getLibrary(id int32, path string) ([]model.Library, error) {
-	stmt := SELECT(
-		Library.AllColumns,
-	).FROM(
-		Library.Table,
-	)
+	stmt := database.QB().From("library").Prepared(true)
 
 	if path == "" {
-		stmt = stmt.WHERE(Library.ID.EQ(Int32(id)))
+		stmt = stmt.Where(goqu.Ex{
+			"id": id,
+		})
 	} else {
-		stmt = stmt.WHERE(Library.ID.EQ(Int32(id)).OR(Library.Path.EQ(String(path))))
+		stmt = stmt.Where(goqu.ExOr{
+			"id": id,
+			"path": path,
+		})
 	}
 
 	var libraries []model.Library
-	err := stmt.Query(db(), &libraries)
+	err := stmt.ScanStructs(&libraries)
+
 	return libraries, err
 }
 
 // newLibrary creates a new library to the database.
 func newLibrary(id int32, path string, layout string) error {
-	stmt := Library.INSERT(Library.ID, Library.Path, Library.Layout).VALUES(id, path, layout).
-		ON_CONFLICT(Library.ID).
-		DO_UPDATE(SET(Library.Path.SET(String(path)), Library.Layout.SET(String(layout))))
-
-	_, err := stmt.Exec(db())
-	if err != nil {
-		log.Error(err)
-	}
+	// does not handle the scenario when the ID exists in db
```

Comment:
In cases like this we probably have to write different logic for all 3 DBMS if there's no one universal abstraction in goqu. In this case for the ON_CONFLICT + DO_UPDATE combination.

Reply:
The first return value of `Exec()` is a `sql.Result` that yields an error if there was any (in the `Err()` method), how about handling this and updating the table as needed?

Reply:
True. In this case that works well.